### PR TITLE
Allowing hooking into the oidc-client events in the AuthenticationProvider

### LINKED ISSUE #415: Proposal - Allowing hooking into the oidc-client events in the AuthenticationProvider
# Scenario

I would like to hook into the events raised by `oidc-client` for logging and specifically the `userLoaded` and `userUnloaded` events for setting and clearing the default authorization header for all `axios` requests.

Although the `useReactOidc` hook exposes `events`, if you try the code below and login, the `addUserLoaded` event is never fired as it's too late in the lifecycle.

```js
<AuthenticationProvider
  configuration={oidcConfiguration}
  loggerLevel={oidcLog.DEBUG}
>
  <DebugEvents />
</AuthenticationProvider>

const DebugEvents = ({ children }) => {
  const { events } = useReactOidc()
  const addUserLoaded = () => console.log('===== userLoaded')

  React.useEffect(() => {
    events.addUserLoaded(addUserLoaded)

    return () => {
      events.removeUserLoaded(addUserLoaded)
    }
  })

  return children
}
```

# Proposal

My proposal is that `AuthenticationProvider` exposes a prop for each of the `oidc-client` events.

For example

```js
<AuthenticationProvider
  configuration={oidcConfiguration}
  loggerLevel={oidcLog.DEBUG}
  onUserLoaded={user => console.log(user)}
  onUserUnloaded={() => console.log('user unloaded')}
  onSilentRenewError={() => console.error('silent renew failed')}
>

</AuthenticationProvider>
```

If there is a way to achieve the above without modifying the library, I'd be happy to hear it.

### PR DESCRIPTION
**Pull request by michael-wolfenden**
Closes #415

### CODE REVIEW

File: packages/context/src/Context/AuthenticationContext.container.js [hunk 1]
```diff
@@ -74,10 +86,44 @@ const AuthenticationProviderInt = ({
...
+    addOidcEvents(
+      oidcState.userManager.events,
+      dispatch,
+      oidcState.userManager,
+      otherProps.onUserLoaded,
+      otherProps.onUserUnloaded,
+      otherProps.onAccessTokenExpiring,
+      otherProps.onAccessTokenExpired,
+      otherProps.onSilentRenewError,
+      otherProps.onUserSignedOut
```
Comment (youf-olivier) [inline packages/context/src/Context/AuthenticationContext.container.js:99]: Can you put all customEvents in one prop ? And add it in the props list of the component

```suggestion
     customEvents
```

**Comment by youf-olivier** [comment]
Thanks for the PR :)

Can you just rebase your version with our lasts PR or resolve conflicts :)

Thanks

**Comment by michael-wolfenden** [comment]
I have incorporated all the current changes in master and added a `customEvents` prop instead of passing the functions individually,

**Comment by miszo** [comment]
Hi, this looks awesome and I would really appreciate if you will introduce it in next release :)

**Comment by youf-olivier** [comment]
After the last refactoring, it seems you have to change some things. Sorry for that :/

**Comment by youf-olivier** [comment]
after that we'll merge your PR.

**Pull request closed without merging**